haystack/preview/components/preprocessors/document_language_classifier.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import List, Dict, Any, Optional
+
+from haystack.preview import component, default_from_dict, default_to_dict, Document
+from haystack.preview.lazy_imports import LazyImport
+
+logger = logging.getLogger(__name__)
+
+with LazyImport("Run 'pip install langdetect'") as langdetect_import:
+    import langdetect
+
+
+@component
+class DocumentLanguageClassifier:
+    """
+    Routes documents onto different output connections depending on their language.
+    This is useful for routing documents to different models in a pipeline depending on their language.
+    The set of supported languages can be specified.
+    For routing texts based on their language use the related TextLanguageClassifier component.
+
+    Example usage in and indexing pipeline that writes only English language documents to a Store:
+
+    ```python
+    document_store = MemoryDocumentStore()
+    p = Pipeline()
+    p.add_component(instance=TextFileToDocument(), name="text_file_converter")
+    p.add_component(instance=DocumentLanguageClassifier(), name="language_classifier")
+    p.add_component(instance=TextDocumentSplitter(), name="splitter")
+    p.add_component(instance=DocumentWriter(document_store=document_store), name="writer")
+    p.connect("text_file_converter.documents", "language_classifier.documents")
+    p.connect("language_classifier.documents", "splitter.documents")
```

Comment:
Shouldn't this be `p.connect("language_classifier.en", "splitter.documents")`?
Also, do we need the splitter in this example pipeline? If not I'd remove it so users can focus on this component.

Reply:
We don't need the splitter in this example, that's right. I'll remove it.